### PR TITLE
`General`: Add client tests for job creation form

### PR DESCRIPTION
**Pull request by ani-ib**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

This PR strengthens the test coverage for the client-side Job Creation Form component.
It ensures that all major form interactions, auto-save behavior, navigation logic, and publish flows are correctly tested — improving maintainability and confidence in future refactors.

### Description

Added comprehensive client-side Vitest suite for
src/main/webapp/app/job/job-creation-form/job-creation-form.component.ts.
- Mocked all key services (AccountService, JobResourceApiService, ToastService, Router, Location, ActivatedRoute) for full isolation.
- Covered key scenarios:
- ✅ Form creation and validation logic
- ✅ Auto-save success and failure states
- ✅ Job publishing flow with privacy checks
- ✅ Back navigation and routing edge cases
- ✅ Stepper button behavior (onClick, disabled conditions)
- ✅ Job DTO creation defaults and trimming behavior

### Steps for Testing

Prerequisites:

1. Checkout this branch
2. Run npm run test:ci
3. Confirm all tests are successful

#### Code Review

- [ ] Code Review 1

#### Manual Tests

- [ ] Test 1

### Screenshots
image

### CODE REVIEW

File: src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+      .overrideComponent(JobCreationFormComponent, {
+        remove: { providers: [JobResourceApiService] },
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(JobCreationFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
```
Comment (Kiara65) [inline src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts:83]: Can you also add an afterEach which restores all mocks to ensure that tests remain isolated and do not affect each other?

```
afterEach(() => {
  vi.restoreAllMocks();
});
```

If I'm correct, you can then remove the vi.clearAllMocks() calls as well.

File: src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+      requirements: 'reqs',
+    });
+    component.additionalInfoForm.patchValue({ privacyAccepted: true });
+
+    component.basicInfoForm.updateValueAndValidity();
+    component.positionDetailsForm.updateValueAndValidity();
+    fixture.detectChanges();
+
+    component.jobId.set('id123');
+    await component.publishJob();
```
Comment (Kiara65) [inline src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts:179]: You're using this code in two tests(should publish successfully and navigate & should handle publishJob failure), would it make sense to create a new method for it, also to allow tests in general to create a valid form?

**Review by Kiara65 — CHANGES_REQUESTED** [review]
Tests are working locally, thank you 😄 I just added two more comments.

File: src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts (hunk 1 above)
Comment (sachmii) [reply src/test/webapp/app/job/job-creation-form/job-creation-form.component.spec.ts:83]: Maybe it would also make sens to clean up the fixtures after each tests.

**Review by sachmii — APPROVED** [review]
Tests look good, only a small comment which I left on Kiara's comment

**Comment by ani-ib** [comment]
Thanks for the feedback @Kiara65 @sachmii I have implemented both of your changes :) Please have another look when you have time

**Comment by sachmii** [comment]
Looks good, thx!

**Review by az108 — APPROVED** [review]
